traffic_monitor/health/cache.go

```diff
@@ -152,6 +156,7 @@ func EvalCache(result cache.ResultInfo, resultStats *threadsafe.ResultStatValHis
 func CalcAvailability(results []cache.Result, pollerName string, statResultHistory *threadsafe.ResultStatHistory, mc tc.TrafficMonitorConfigMap, toData todata.TOData, localCacheStatusThreadsafe threadsafe.CacheAvailableStatus, localStates peer.CRStatesThreadsafe, events ThreadsafeEvents, protocol config.PollingProtocol) {
```

Comment:
in health/cache.go  in CalcAvailability  what are your thoughts on ending this for loop 
`for interfaceName, _ := range result.Interfaces() {`
earlier right before this line
```
// Compute aggregate data based on each interface
			aggregateStatus := cache.AvailableStatus{
```
and then this last line
`			localStates.SetCache(tc.CacheName(result.ID), tc.IsAvailable{IsAvailable: newAvailableState, Ipv4Available: availableTuple.IPv4, Ipv6Available: availableTuple.IPv6})`
should really use the aggregate status right? something like
```
			aggregateAvailableState := processAvailableTuple(aggregateStatus.Available, serverInfo)
			localStates.SetCache(tc.CacheName(result.ID), tc.IsAvailable{IsAvailable: aggregateAvailableState, Ipv4Available: aggregateStatus.Available.IPv4, Ipv6Available: aggregateStatus.Available.IPv6})
```
that way it doesnt calculate the aggregate stats for each loop through the interfaces and the local state of the cache is based on the aggregate instead of on just the last interface it found?

and then this part
```
			if available, ok := localStates.GetCache(tc.CacheName(result.ID)); !ok || available.IsAvailable != newAvailableState {
				protocol := "IPv4"
				if !usingIPv4 {
					protocol = "IPv6"
				}
				log.Infof("Changing state for %s was: %t now: %t because %s poller: %v on protocol %v error: %v", result.ID, available.IsAvailable, newAvailableState, whyAvailable, pollerName, protocol, result.Error)
				events.Add(Event{Time: Time(time.Now()), Description: "Protocol: (" + protocol + ") " + whyAvailable + " (" + pollerName + ")", Name: string(result.ID), Hostname: string(result.ID), Type: toData.ServerTypes[tc.CacheName(result.ID)].String(), Available: newAvailableState, IPv4Available: availableTuple.IPv4, IPv6Available: availableTuple.IPv6})
			}
```
should be moved to the bottom and based on the aggregate as well?

Reply:
You are correct, originally I thought we were going to need to break out localStates by interface but that is not the case.